tools/lsp/server/parser/yaml_parser.go

```diff
@@ -0,0 +1,145 @@
+package parser
+
+import (
+	"strings"
+
+	"github.com/go-logr/logr"
+	protocol "github.com/tliron/glsp/protocol_3_16"
+	"gopkg.in/yaml.v3"
+)
+
+// YAMLParser handles parsing and analysis of YAML documents for the KRO LSP server.
+// It provides YAML syntax validation and KRO resource detection capabilities,
+// serving as the first stage in the document validation pipeline.
+type YAMLParser struct {
+	logger logr.Logger
+}
+
+// ParseResult represents the outcome of YAML parsing operations.
+// It contains the parsed YAML node structure, any parsing diagnostics,
+// and an error status for use by the LSP validation pipeline.
+type ParseResult struct {
+	Node        *yaml.Node
+	Diagnostics []protocol.Diagnostic
+	HasErrors   bool
+}
+
+// NewYAMLParser creates a new YAML parser instance with the provided logger.
+// The parser uses yaml.v3 for enhanced node positioning and error reporting capabilities.
+func NewYAMLParser(logger logr.Logger) *YAMLParser {
+	return &YAMLParser{
+		logger: logger,
+	}
+}
+
+// Parse performs YAML syntax validation and parsing on the provided content.
+// Returns a ParseResult containing either the parsed AST or diagnostic errors.
+// This is the primary entry point for YAML validation in the LSP pipeline.
+func (p *YAMLParser) Parse(content string) ParseResult {
+	// Initialize empty result
+	result := ParseResult{
+		Diagnostics: []protocol.Diagnostic{},
+		HasErrors:   false,
+	}
+
+	// Skip parsing for empty or whitespace-only content
+	if strings.TrimSpace(content) == "" {
+		return result
+	}
+
+	// Parse YAML content into AST using yaml.v3
+	// yaml.v3 provides better error reporting and node positioning than yaml.v2
+	var node yaml.Node
+	err := yaml.Unmarshal([]byte(content), &node)
```

Comment:
nit: switch to k8s yaml here

Reply:
hmm, youre right, I believe for your use case you are fine, since your yaml is never actually any k8s related YAML AFAIK. feel free to just resolve this!